lib/SPIRV/OCLTypeToSPIRV.cpp

```diff
@@ -294,7 +294,7 @@ OCLTypeToSPIRV::adaptArgumentsByMetadata(Function* F) {
     auto NewTy = *PI;
     if (OCLTyStr == OCL_TYPE_NAME_SAMPLER_T && !NewTy->isStructTy()) {
       addAdaptedType(Arg, getOrCreateOpaquePtrType(M,
-        getSPIRVTypeName(kSPIRVTypeName::Sampler)));
+        getSPIRVTypeName(kSPIRVTypeName::Sampler), SPIRAS_Private));
```

Comment:
It makes more sense to put sampler in constant addr space since they never change.

Also, we better factor out sampler type creation as a function since it is used many times.